src/main/scala/com/lucidchart/open/relate/ColReader.scala

```diff
@@ -0,0 +1,76 @@
+package com.lucidchart.open.relate
+
+import java.nio.ByteBuffer
+import java.sql.ResultSet
+import java.util.{Date, UUID}
+import shapeless._
+
+class NullColumnException(col: String) extends Exception(s"Unexpected null value in column: $col")
+
+trait ColReader[A] { self =>
+  def read(col: String, row: SqlResult): Option[A]
```

Comment:
We could consider also having a method for column position.

```scala
def read(col: Int, row: SqlResult)
```

If we did that, we could have

```scala
object Parseable {
  implicit def tuple2[A : ColReader, B : ColReader]: Parseable[(A, B)] =
    Parseable(row => (implicitly[ColReader[A]].read(0, row), implicitly[ColReader[B]].read(1, row))
}
```

And then

```scala
// get map from trainer id to number of pokemon
sql"SELECT trainer_id, COUNT(*) FROM pokemon GROUP BY trainer_id".Map[Int, Int]
```